urlcdd0c1ab49_emiln_JHouseFireExplorer_tgz-pJ8-dk_brics_automaton_DatatypesJ8/src/dk/brics/automaton/ShuffleOperations.java

```diff
@@ -0,0 +1,338 @@
+		Transition[][] a_transitions = Automaton.getSortedTransitions(a.getStates());
+		TransitionComparator tc = new TransitionComparator(false);
+		ShuffleConfiguration init = new ShuffleConfiguration(ca, a);
+		LinkedList<ShuffleConfiguration> pending = new LinkedList<ShuffleConfiguration>();
+		Set<ShuffleConfiguration> visited = new HashSet<ShuffleConfiguration>();
+		pending.add(init);
+		visited.add(init);
+		while (!pending.isEmpty()) {
+			ShuffleConfiguration c = pending.removeFirst();
+			boolean good = true;
+			for (int i1 = 0; i1 < ca.size(); i1++)
+				if (!c.ca_states[i1].accept) {
+					good = false;
+					break;
+				}
+			if (c.a_state.accept)
+				good = false;
+			if (good) {
+				StringBuilder sb = new StringBuilder();
+				while (c.prev != null) {
+					sb.append(c.min);
+					c = c.prev;
+				}
+				StringBuilder sb2 = new StringBuilder();
+				for (int j = sb.length() - 1; j >= 0; j--)
+					sb2.append(sb.charAt(j));
+				return sb2.toString();
+			}
+			Transition[] ta2 = a_transitions[c.a_state.number];
+			for (int i1 = 0; i1 < ca.size(); i1++) {
+				if (c.shuffle_suspended)
+					i1 = c.suspended1;
+				loop: for (Transition t1 : ca_transitions[i1][c.ca_states[i1].number]) {
+					List<Transition> lt = new ArrayList<Transition>();
+					int j = Arrays.binarySearch(ta2, t1, tc);
+					if (j < 0)
+						j = -j - 1;
+					if (j > 0 && ta2[j - 1].max >= t1.min)
+						j--;
+					while (j < ta2.length) {
+						Transition t2 = ta2[j++];
+						char min = t1.min;
+						char max = t1.max;
+						if (t2.min > min)
+							min = t2.min;
+						if (t2.max < max)
+							max = t2.max;
+						if (min <= max) {
+							add(suspend_shuffle, resume_shuffle, pending, visited, c, i1, t1, t2, min, max);
+							lt.add(new Transition(min, max, null));
+						} else
+							break;
+					}
+					Transition[] at = lt.toArray(new Transition[lt.size()]);
+					Arrays.sort(at, tc);
+					char min = t1.min;
+					for (int k = 0; k < at.length; k++) {
+						if (at[k].min > min)
+							break;
+						if (at[k].max >= t1.max)
+							continue loop;
+						min = (char)(at[k].max + 1);
+					}
+					ShuffleConfiguration nc = new ShuffleConfiguration(c, i1, t1.to, min);
+					StringBuilder sb = new StringBuilder();
+					ShuffleConfiguration b = nc;
+					while (b.prev != null) {
+						sb.append(b.min);
+						b = b.prev;
+					}
+					StringBuilder sb2 = new StringBuilder();
+					for (int m = sb.length() - 1; m >= 0; m--)
+						sb2.append(sb.charAt(m));
+					if (c.shuffle_suspended)
+						sb2.append(BasicOperations.getShortestExample(nc.ca_states[c.suspended1], true));
+					for (i1 = 0; i1 < ca.size(); i1++)
+						if (!c.shuffle_suspended || i1 != c.suspended1)
+							sb2.append(BasicOperations.getShortestExample(nc.ca_states[i1], true));
+					return sb2.toString();
+				}
+				if (c.shuffle_suspended)
+					break;
+			}
+		}
+		return null;
+	}
+
+	private static void add(Character suspend_shuffle, Character resume_shuffle, 
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

The cyclomatic complexity of this method is 24. By comparison, 99% of the methods in the CodeGuru reference dataset have a lower cyclomatic complexity. This indicates the method has a high number of decisions and it can make the logic difficult to understand and test. We recommend that you simplify this method or break it into multiple methods.